core/kazoo_modb/src/kazoo_modb_maintenance.erl

```diff
@@ -86,67 +90,112 @@ do_archive_modbs(MODbs, AccountId) ->
+    {Year, Month, _} = erlang:date(),
+    verify_rollups(Account, Year, Month, Options).
 
-verify_rollups(AccountDb, Year, Month) ->
+-spec verify_rollups(ne_binary(), kz_year(), kz_month(), kz_proplist()) -> 'ok'.
+verify_rollups(AccountDb, Year, Month, Options) ->
+    AllRollups = props:get_value('all_rollups', Options),
     AccountId = kz_util:format_account_id(AccountDb, 'raw'),
     case kazoo_modb:open_doc(AccountDb, <<"monthly_rollup">>, Year, Month) of
+        {'ok', JObj} when AllRollups ->
+            _ = verify_rollups(AccountDb, Year, Month, AccountId, JObj),
+            {PYear, PMonth} = kazoo_modb_util:prev_year_month(Year, Month),
+            verify_rollups(AccountDb, PYear, PMonth, Options);
         {'ok', JObj} ->
             verify_rollups(AccountDb, Year, Month, AccountId, JObj);
         {'error', 'not_found'} ->
-            io:format("    account ~s : no modb @ ~p-~p~n", [AccountId, Year, Month]);
+            io:format("    account ~s has no rollup for ~p-~p~n", [AccountId, Year, Month]);
         Else ->
-            io:format("    account ~s: error getting monthly rollup ~p~n", [AccountId, Else])
+            io:format("    account ~s error getting monthly rollup ~p~n", [AccountId, Else])
     end.
 
+-spec verify_rollups(ne_binary(), kz_year(), kz_month(), ne_binary(), kz_json:object()) -> 'ok'.
 verify_rollups(AccountDb, Year, Month, AccountId, JObj) ->
+    {PYear, PMonth} = kazoo_modb_util:prev_year_month(Year, Month),
     Balance = wht_util:previous_balance(AccountDb
-                                       ,kz_util:to_binary(Year)
-                                       ,kz_util:pad_month(Month)
+                                       ,kz_util:to_binary(PYear)
+                                       ,kz_util:pad_month(PMonth)
                                        ),
     case rollup_balance(JObj) of
         Balance ->
-            io:format("    account ~s rollup for ~p-~p confirmed~n", [AccountId, Year, Month]),
-            {PYear, PMonth} = kazoo_modb_util:prev_year_month(Year, Month),
-            verify_rollups(AccountDb, PYear, PMonth);
+            io:format("    account ~s rollup for ~p-~p confirmed~n", [AccountId, Year, Month]);
         _RollupBalance ->
-            io:format("    account ~s has a discrepancy! rollup/balance ~p/~p~n"
-                     ,[AccountId, _RollupBalance, Balance]
+            io:format("    account ~s rollup for ~p-~p balance is ~p but previous month balance was ~p~n"
+                     ,[AccountId, Year, Month, _RollupBalance, Balance]
                      )
     end.
 
 -spec fix_rollup(ne_binary()) -> 'ok'.
 fix_rollup(Account) ->
+    {Year, Month, _} = erlang:date(),
+    fix_rollup(Account, Year, Month).
+
+-spec fix_rollup(ne_binary(), ne_binary(), ne_binary()) -> 'ok'.
+fix_rollup(Account, Year, Month) ->
     AccountId = kz_util:format_account_id(Account, 'raw'),
-    {Y, M, _} = erlang:date(),
+    Y = kz_util:to_integer(Year),
```

Comment:
no need for convert to integer since `kazoo_modb_util:prev_year_month` is already doing that